sheep/src/lib.rs

```diff
@@ -38,12 +46,22 @@ pub fn pack<P: Packer>(
     stride: usize,
     options: P::Options,
 ) -> Vec<SpriteSheet> {
+    let mut hashes: HashMap<&[u8], usize, BuildHasherDefault<XxHash64>> = Default::default();
+    let mut aliases: Vec<SmallVec<[usize; 1]>> =
+        (0..input.len()).map(|_| SmallVec::new()).collect();
```

Comment:
I'd say just do a `HashMap::with_capacity(input.len())` here. You won't have unnecessary allocations, but you don't need to filter for empty `SmallVec`s later.

See below too